Big_Integer/main.cpp

```diff
@@ -0,0 +1,476 @@
+        int to = -1;
+        for (int i = 0; ; i ++) {
+            if ((1 << i) >= length_A && (1 << i) >= length_B) {
+                to = (1 << i);
+                break;
+            }
+        }
+        if (length_A < to) {
+            reverse(a.begin(), a.end());
+            while (length_A < to) {
+                a += '0';
+                length_A ++;
+            }
+            reverse(a.begin(), a.end());
+        }
+        if (length_B < to) {
+            reverse(b.begin(), b.end());
+            while (length_B < to) {
+                b += '0';
+                length_B ++;
+            }
+            reverse(b.begin(), b.end());
+        }
+
+    }
+    string defaultMultiplier(string &a, string &b) { // builtin multiplication
+        int aValue = stoi(a); //string to int
+        int bValue = stoi(b);
+        return to_string(1LL * aValue * bValue);
+    }
+
+    string multiply(string a, string b) {     //Day 5: Using karatsuba Algorithm...
+        if ((int)a.size() < 9 && (int)b.size() < 9) {
+            return defaultMultiplier(a, b);
+        }
+
+        normalize(a, b);
+        string leftA, rightA;
+        string leftB, rightB;
+        _break(a, leftA, rightA);
+        _break(b, leftB, rightB);
+        string p1 = multiply(leftA, leftB);
+        string p2 = multiply(rightA, rightB);
+        string a_sum = add(leftA, rightA);
+        string b_sum = add(leftB, rightB);
+        string p3 = multiply(a_sum, b_sum);
+        string delta31 = subtract(p3, p1);
+        string p4 = subtract(delta31, p2);
+        insertTrailingZeros(p1, (int)a.size());
+        insertTrailingZeros(p4, (int)a.size() >> 1);
+        string sum= add(p2, p4);
+        return add(p1, sum);
+    }
+
+
+
+    string _modulo(string a, string b) { // modulo operation of two integers(strings)
+        string result = "";
+        string current = "";
+        for (char ch: a) {
+            current += ch;
+            if (_compare(current, b) != -1) {
+                int wheel = 0;
+                while (_compare(current, b) != -1) {
+                    wheel ++;
+                    current = subtract(current, b);
+                }
+                result += (char)(wheel + '0');
+            }
+        }
+        reverse(current.begin(), current.end());
+        while ((int)current.size() > 1 && current.back() == '0') {
+            current.pop_back();
+        }
+        reverse(current.begin(), current.end());
+        return current;
+    }
+    string fast_modulo(string a, int b) { // faster modulo function
+        string result = "";
+        int current = 0;
+        for (char ch: a) {
+            current = (current << 1) + (current << 3) + ch - '0';
+            if (current >= b) {
+                result += (char)(current / b + '0');
+                current %= b;
+            }
+        }
+        return to_string(current);
+    }
+
+    string modulo(string a, string b) { // Day 5: modulo of two integers
```

Comment:
Codacy found an issue: [Function parameter 'a' should be passed by const reference.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6290669)